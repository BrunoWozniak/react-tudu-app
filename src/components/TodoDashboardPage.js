import React from 'react';
import { connect } from 'react-redux';

import TodoList from './TodoList';
import TodoListFilters from './TodoListFilters';
import TodoSummary from './TodoSummary';
import { startSetTodos } from '../actions/todos';


class TodoDashboardPage extends React.Component {
	componentWillMount() {
		this.props.startSetTodos();
	}

	render () {
		return (
			<div>
				<TodoSummary />
				<TodoListFilters />
				<TodoList />
			</div>
		);
	}
}
  
const mapDispatchToProps = (dispatch) => ({
	startSetTodos: () => dispatch(startSetTodos())
});

export default connect(null, mapDispatchToProps)(TodoDashboardPage);
