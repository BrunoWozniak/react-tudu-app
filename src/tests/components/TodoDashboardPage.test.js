import React from 'react';
import { shallow } from 'enzyme';
import TodoDashboardPage from '../../components/TodoDashboardPage';

test('should render TodoDashboardPage correctly', () => {
    const wrapper = shallow(<TodoDashboardPage />);
    expect(wrapper).toMatchSnapshot();
});