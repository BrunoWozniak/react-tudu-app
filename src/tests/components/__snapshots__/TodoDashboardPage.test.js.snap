// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodoDashboardPage correctly 1`] = `
<div>
  <Connect(TodoSummary) />
  <Connect(TodoListFilters) />
  <Connect(TodoList) />
</div>
`;
